index.js

```diff
@@ -144,7 +144,10 @@ class Discovery extends EventEmitter {
     super()
 
     if (!opts) opts = {}
-
+    // if ephemeral is undefined, null or anything other than a boolean
+    // then this signifies adaptive ephemerality
+    opts.adaptive = typeof opts.ephemeral !== 'boolean'
```

Comment:
can we only set the default if it's not set?

Reply:
good point, fixed